lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+  return success();
+}
+
+LogicalResult translateAllocToSDFG(AllocOp &op, JsonEmitter &jemit) {
+  if (op.getType().getShape().size() == 0)
+    return printScalar(op, jemit);
+
+  jemit.startNamedObject(op.getName());
+  jemit.printKVPair("type", "Array");
+
+  jemit.startNamedObject("attributes");
+  // TODO: Print the values you can derive
+  // jemit.printKVPair("allow_conflicts", "false", /*stringify=*/false);
+
+  jemit.startNamedList("strides");
+  ArrayRef<int64_t> shape = op.getType().getIntegers();
+
+  for (int i = shape.size() - 1; i > 0; --i) {
+    jemit.startEntry();
+    jemit.printInt(shape[i]);
+  }
+
+  jemit.startEntry();
+  jemit.printInt(1);
+  jemit.endList(); // strides
+
+  // jemit.printKVPair("total_size", "4");
+  jemit.startNamedList("offset");
+  for (unsigned i = 0; i < shape.size(); ++i) {
+    jemit.startEntry();
+    jemit.printInt(0);
+  }
+  jemit.endList(); // offset
+
+  // jemit.printKVPair("may_alias", "false", /*stringify*/false);
+  // jemit.printKVPair("alignment", 0, /*stringify*/false);
+  Type element = op.getType().getElementType();
+  if (translateTypeToSDFG(element, op.getLoc(), jemit, "dtype").failed())
+    return failure();
+
+  jemit.startNamedList("shape");
+  for (int64_t s : shape) {
+    jemit.startEntry();
+    jemit.printInt(s);
+  }
+  jemit.endList(); // shape
+
+  jemit.printKVPair("transient", "false", /*stringify=*/false);
+  jemit.printKVPair("storage", "Default");
+  jemit.printKVPair("lifetime", "Scope");
+  // jemit.startNamedObject("location");
+  // jemit.endObject(); // location
+
+  // jemit.printKVPair("debuginfo", "null", /*stringify=*/false);
+
+  jemit.endObject(); // attributes
+  jemit.endObject();
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// AllocTransientOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateAllocTransientToSDFG(AllocTransientOp &op,
+                                            JsonEmitter &jemit) {
+  AsmState state(op.getParentSDFG());
+  std::string name;
+  llvm::raw_string_ostream nameStream(name);
+  op->getResult(0).printAsOperand(nameStream, state);
+
+  jemit.startNamedObject(name);
+  jemit.printKVPair("type", "Array");
+
+  jemit.startNamedObject("attributes");
+  // TODO: Print the values you can derive
+  // jemit.printKVPair("allow_conflicts", "false", /*stringify=*/false);
+  // jemit.startNamedList("strides");
+  // jemit.endList(); // strides
+
+  // jemit.printKVPair("total_size", "4");
+  // jemit.startNamedList("offset");
+  // jemit.endList(); // offset
+
+  // jemit.printKVPair("may_alias", "false", /*stringify*/false);
+  // jemit.printKVPair("alignment", 0, /*stringify*/false);
+  jemit.printKVPair("dtype", "int32");
+  // jemit.startNamedList("shape");
+  // jemit.endList(); // shape
+
+  jemit.printKVPair("transient", "true", /*stringify=*/false);
+  jemit.printKVPair("storage", "Default");
```

Comment:
Same as above (if this attribute is given, I wouldn't want the translator to override it)